src/state/application/updater.ts

```diff
@@ -119,7 +121,14 @@ export default function Updater(): null {
   }, [dispatch, debouncedState.chainId])
 
   useEffect(() => {
-    if (!account || !library?.provider?.request || !library?.provider?.isMetaMask) {
+    const isCbWalletDappBrowser = window?.ethereum?.isCoinbaseWallet
+    const isWalletlinkOrInjectedConnector =
+      connector instanceof WalletLinkConnector || connector instanceof InjectedConnector
+    const isWalletlink =
+      !!window?.WalletLinkProvider || (!!window?.walletLinkExtension && isWalletlinkOrInjectedConnector)
```

Comment:
is `window?.WalletLinkProvider` not redundant with `connector instanceof WalletLinkConnector`?

Reply:
good call. updated to simplify.